File: OpenWeatherAPI/OpenWeatherAPI.Models/Models/CityDTO.cs
using OpenWeatherAPI.Business.Base;
using System;
using System.Collections.Generic;
using System.Text;

namespace OpenWeatherAPI.Business.Models
{
    public class CityDTO : BaseEntity
    {
        public string Name { get; set; }
        public string State { get; set; }
        public double Longitude { get; set; }
        public double Latitude { get; set; }
        public int CountryId { get; set; }
        public CountryDTO Country { get; set; }
        public BranchOfficeDTO BranchOffice { get; set; }
        //public ICollection<WeatherConditionDTO> WeatherCondition { get; set; }

    }
}

File: OpenWeatherWebApp/src/app/shared/DTOs/CityDTO.ts
import { BranchOfficeDTO } from './BranchOfficeDTO';
import {CountryDTO} from './CountryDTO';

export class CityDTO
{
    constructor(
        public id: number,
        public name:string,
        public state:string,
        public longitude:number,
        public latitude:number,
        public countryId:number,
        public country:CountryDTO,
        public branchOffice:BranchOfficeDTO
    ){}
    
}